src/e2e/java/teammates/e2e/util/BackDoor.java

```diff
@@ -282,6 +283,47 @@ public static CourseData getCourseData(String courseId) {
      */
     public static CourseAttributes getCourse(String courseId) {
         CourseData courseData = getCourseData(courseId);
+        if (courseData == null) {
+            return null;
+        }
+        return CourseAttributes.builder(courseData.getCourseId())
+                .withName(courseData.getCourseName())
+                .withTimezone(ZoneId.of(courseData.getTimeZone()))
+                .build();
+    }
+
+    /**
+     * Gets archived course data from the datastore.
+     */
+    public static CourseData getArchivedCourseData(String instructorId, String courseId) {
```

Comment:
Is this method really needed? It looks like it is hitting the same endpoint as `getCourse`.

Reply:
It is actually a different endpoint as archived status is not saved in the CourseAttributes but rather in InstructorAttributes. Hence the GetCourseAction endpoint cannot be used and the getCourses endpoint which gets all courses by instructor id has to be used instead.